pkg/controllers/monitoring-stack/components.go

```diff
@@ -21,7 +21,7 @@ const AdditionalScrapeConfigsSelfScrapeKey = "self-scrape-config"
 
 type emptyObjectFunc func() client.Object
 
-type patchObjectFunc func(object client.Object) (client.Object, error)
+type patchObjectFunc func(existing client.Object) (client.Object, error)
```

Comment:
I was also thinking patch should just mutate the `existing` and return error if any .. so that the caller only needs to 
```
if err := pather.patch(existing); err != nil {
  return err
}
```

but it will require modification of `serviceAccount` patcher. 
Additionally, the patch will change to 
```go
patch: func (existing client.Object) error {
  if existing == nil {
    return errNilObject
  }
  actual, ok := existing.(v1.ProperType)
  if !ok {
    return errTypeMismatch
  }
 
 desired := newObjectForMs(ms)
  current.Spec = desired.Spec
  return nil
}
```